frontend/src/pages/game.tsx

```diff
@@ -0,0 +1,112 @@
+// Game page that is shown when the user goes to localhost:3000/game
+
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { io } from 'socket.io-client';
+import GameControl from '../components/Game/GameControl.tsx';
+
+const Game = () => {
+	const [socket, setSocket] = useState<Socket | null>(null);
+	const [tempToken, setTempToken] = useState<string | null>(null); //tijdelijke oplossing voor Token
+	const [games, setGames] = useState<Match[] | null>([]);
+	const [selectedGame, setSelectedGame] = useState<number | null>(null);
+    
+    useEffect(() => {
+        //because dev mode sometimes didnt disconnect old sockets
+        if (socket) {
+			socket.disconnect(); // Disconnect existing socket if any
+            console.log('Previous socket disconnected');
+        }
+        
+        // Initialize socket connection
+		const token: string = localStorage.getItem('token');
+        const socketIo: Socket = io('http://localhost:3001/game', {
+			transports: ['websocket', 'polling'],
+            query: { token: token } // Hier de token uit localstorage halen
+        });
+
+		//temporary replacing token for websocketID for testing
+        socketIo.on('token', (websocketID: string) => {
+            setTempToken(websocketID);
+            console.log('replaced token with websocketID')
+        })
+		
+        // Set socket instance in state
+        setSocket(socketIo);
+
+		const fetchGames = async () => {
+            try {
+                const response: AxiosResponse<Match[]> = await axios.get(`http://localhost:3001/game/matches`);
+                setGames(response.data);
+            } catch (error) {
+                console.error('Error fetching games:', error);
+            }
+        };
+    
+        fetchGames();
+		
+		socketIo.on('newGame', (data: any) => {
```

Comment:
kijk, dan kan je hier on('newGame', () => fetchGames() callen en dan is het veel overzichtelijker (zie mijn comment in de game.gateway.ts (line 50))